scene/gui/range.h

```diff
@@ -41,6 +41,7 @@ class Range : public Control {
 		double min = 0.0;
 		double max = 100.0;
 		double step = 1.0;
+		bool skip_step_rounding = false;
```

Comment:
I'm uncertain if `Shared` is the correct place or the class Range itself. But I think it should be possible (at the very least internally), to prevent rounding.

Reply:
Sorry for the constant nagging without doing any actual work myself, but I think this new `set_disallow_step_rounding()` method has the same effect as simply doing `set_step(0.0)`? 😅

Reply:
It does have the same effect.

With the additional bonus of breaking the SpinBox: the increment/decrement will stop working.  
With this change, it will keep working and increments/decrements in seconds (which is not ideal, tbh, but first I want to see if it actually works, then we can talk about the steps. Or even use the current steps).

If this works out, I might need to split the PR anyway, because I'm touching GUI stuff… (but first testing if it works)

Reply:
Ah, that's a good point! Yes, I didn't think about the step buttons. So this way, we could have a SpinBox that supports step increments but still allows floating point entries at full precision. Yes, I think this would be one way to work around the original problem of not being able to enter the precise "seconds" equivalent of the sample count.

Reply:
At first, I wanted to set negative steps to prevent auto-rounding. But then the buttons would be inversed :laughing:

Reply:
Hey, I tested the updated PR, I'm afraid the issue still there. 🙁

The spinbox itself is working fine, though it still rounds values with more than six digits after the decimal dot (but that's more than enough for the loop_offset issue).

However, on reimport the rounding issue persists.

Reply:
Bummer. Hm… I'll have a closer look over the weekend and ask in the core channel (sounds like the best fit because of the serializing)

Reply:
You can set separate arrow step for SpinBox. The new property is unnecessary.

Reply:
@KoBeWi: This PR will be changed (see https://github.com/godotengine/godot/pull/87882#issuecomment-1938766979). I haven't received any response yet, so I'll change the code as stated there.

And I completely forgot that we still run into rounding issues, even if double (de-)serialization would be fully exhausted. It just happens at a later decimal place. So rounding should happen either way.